### ISSUE TITLE
test a new install before app (with branch-io) is released to the store?

### ISSUE THREAD

**Issue by pcompassion** [BranchMetrics/iOS-Deferred-Deep-Linking-SDK#246]
I'd like to test

```
onlyOnce = true;
if installParam has a specific param && onlyOnce {
   // do something
   onlyOnce = false;
}
```

when user clicks a link, if he doesn't  have an app installed, he'll be redirected to the app store.
The app on app store doesn't have the above code yet, because I'm integrating Branch.io and previous version didn't have branch.io at all.

So he has to have new version of app installed, and I need a way to fool branch.io that it got some install parameter when it was installed. 

How can I do that?

**Comment by aaustin** [comment]
This is a great suggestion. I've been wanted to add a 'deep link 100% of time mode' to the SDKs for a while. Let me get it added this sprint.

In the mean time, unfortunately, here's the test flow:
0. Uninstall the app
1. Click a Branch link with the deep link data you want (here's a couple ways you can create one https://dev.branch.io/link_creation_guide/)
2. Install the app from Xcode with proper Branch key
3. Data will be passed through to initSession callback

**Comment by aaustin** [comment]
Closing issue for now.

**Comment by pcompassion** [comment]
Thank you for the quick answer.

By the way, on production, I'm creating a link programmatically in the app and a user will give the link to other users through another messenger app (eg, whatsapp).  
The link would be of the form `http://bnc.lt/xxxx`

Does your methodology still apply in this scenario?

It sounds like branch.io looks at the data browser stores (How would initSession would fetch link data that was clicked outside of myapp?)

In other words, would branch.io work when someone clicks the link in some arbitrary app? 
(The exact app I'm using is kakao talk messenger app: https://itunes.apple.com/us/app/kakaotalk/id362057947?mt=8)

**Comment by aaustin** [comment]
Sorry for the delay. 

You know it! We register the click of the link and match it up to the app session on the same device. That's how we pass in the deep link data. In order to test with a dynamically generated link, you can use the same process above - in Kakao and any other apps.

**Comment by pcompassion** [comment]
Thank you!

**Comment by MuhammadAbdullah54321** [comment]
> * all the app from Xcode with proper Branch key

can you please explain what do you mean by point 2 **Install the app from Xcode with proper Branch key
**

**Comment by echo-branch** [comment]
@MuhammadAbdullah54321 
Please open new tickets for any questions, this ticket is almost 7 years old.

`Install the app from Xcode with proper Branch key` just refers to launching your app from Xcode to a device/simulator. Just make sure you've set the correct Branch key during configuration.
